keras_cv/layers/preprocessing3d/crop_objects.py

```diff
@@ -0,0 +1,198 @@
+        The first 5 features are [x, y, z, class, range].
+      bounding_boxes: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of boxes, num of box features].
+        The first 8 features are [x, y, z, dx, dy, dz, phi, box class].
+
+    Output shape:
+      A dictionary of Tensors with the same shape as input Tensors and two additional items for object_point_clouds and OBJECT_BOUNDING_BOXES.
+
+    Arguments:
+      label_index: A int scalar sets the target object index.
+      min_points_per_bounding_boxes: A int scalar sets the min number of points in a bounding box. If a bounding box contains less than min_points_per_bounding_boxes, the bounding box is filtered out.
+      max_points_per_bounding_boxes: A int scalar sets the max number of points in a bounding box. All the object point clouds will be padded or trimmed to the same shape, where the number of points dimension is max_points_per_bounding_boxes.
+    """
+
+    def __init__(
+        self,
+        label_index=0,
+        min_points_per_bounding_boxes=0,
+        max_points_per_bounding_boxes=2000,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        if label_index < 0:
+            raise ValueError("label_index must be >=0.")
+        if min_points_per_bounding_boxes < 0:
+            raise ValueError("min_points_per_bounding_boxes must be >=0.")
+        if max_points_per_bounding_boxes < 0:
+            raise ValueError("max_points_per_bounding_boxes must be >=0.")
+        if min_points_per_bounding_boxes > max_points_per_bounding_boxes:
+            raise ValueError(
+                "max_paste_bounding_boxes must be >= min_points_per_bounding_boxes."
+            )
+
+        self._label_index = label_index
+        self._min_points_per_bounding_boxes = min_points_per_bounding_boxes
+        self._max_points_per_bounding_boxes = max_points_per_bounding_boxes
+        self._iou_3d = iou_3d.IoU3D
+
+    def augment_point_clouds_bounding_boxes(
+        self, point_clouds, bounding_boxes, **kwargs
+    ):
+        if self._label_index > 0:
+            bounding_boxes_mask = tf.math.equal(
+                bounding_boxes[0, :, 7], self._label_index
+            )
+            object_bounding_boxes = tf.boolean_mask(
+                bounding_boxes, bounding_boxes_mask, axis=1
+            )
+        else:
+            bounding_boxes_mask = ~tf.math.equal(bounding_boxes[0, :, 7], 0.0)
+            object_bounding_boxes = tf.boolean_mask(
+                bounding_boxes, bounding_boxes_mask, axis=1
+            )
+
+        points_in_bounding_boxes = is_within_box3d(
+            point_clouds, object_bounding_boxes[:, :, :7]
+        )
+        # Filter bounding boxes using the current frame.
+        min_points_filter = (
+            tf.reduce_sum(tf.cast(points_in_bounding_boxes[0], dtype=tf.int32), axis=0)
+            >= self._min_points_per_bounding_boxes
+        )
+
+        object_bounding_boxes = tf.boolean_mask(
```

Comment:
to me, this is still bounding_boxes, but it's a filtered version of bounding_boxes, no? (I was trying to understand why we have two different formats, one is BOUNDING_BOXES, the other one is OBJECT_BOUNDING_BOXES

Reply:
OBJECT_BOUNDING_BOXES is a subset of BOUNDING_BOXES

Reply:
Yes, it is a filtered version of BOUNDING_BOXES

Reply:
so, can we re-use BOUNDING_BOXES?

Reply:
I think it is better to use different names to distinguish OBJECT_BOUNDING_BOXES from BOUNDING_BOXES. Same for POINT_CLOUDS and OBJECT_POINT_CLOUDS.